### DISCUSSION TITLE
Configuring Sidecar Proxy to use one specific ciper

### DISCUSSION THREAD

**Discussion by marcin-wrzeszcz** [istio/istio#51732] [General]
Hi 

I have a application running on a cluster that utilises sidecar proxy. 

image: docker.io/istio/proxyv2:1.14.1

I am trying to force sidecar proxy to use one specific cipher in order to create small benchmark system.

I have applied changes to 'cipherSuites'. Run traffic through the program and captured the pcap file.

By default - no changes at all - the Cipher Suites in TLS Client Hello message contains 9 suits. 

However  when cipherSuites is changed to a single cipher, examining the pcap I can see that in TLS negotiation that there are always 3 ciphers present and then the cipher I want to test with.

AES_128_GCM_SHA256, AES_256_GCM_SHA384, CHACHA20_POLY1305_SHA256

And AES_128_GCM_SHA256 is always selected in server response. I am guessing hardcoded options in envoy sourcecode

Is there a way around it? Apart from recompiling image with envoy source changes?

**Comment by howardjohn** [comment]
Istio 1.14 has been EOL for many many years, and has serious bugs, security issues, and performance issues. I strongly recommend usage of it for any system.

In addition, the feature to configure this was added after 1.14: https://istio.io/latest/docs/reference/config/istio.mesh.v1alpha1/#MeshConfig-TLSConfig

**Comment by marcin-wrzeszcz** [comment]
Thanks for reply, I will try with the latest version

**Comment by marcin-wrzeszcz** [comment]
I was able to get back to this. 
I have upgraded the Istio Installation to v1.20.0 and as per suggestion I have extended the meshConfig section to include selected cipher

```
  meshConfig:
    accessLogFile: /dev/stdout
    defaultConfig:
      proxyMetadata: {}
    enablePrometheusMerge: true
    tlsDefaults:
      cipherSuites:
      - AES256-GCM-SHA384
```

However that did not force tls server to use specified cipher. In the tls handshake 6 ciphers are offered and aes128 one is always picked.

We inject the sidecar proxy into pods with chart entries,
```
    sidecar.istio.io/inject: "true"
(...)
    image: docker.io/istio/proxyv2:1.20.0
```
